connectors/connectors.go

```diff
@@ -528,7 +530,13 @@ func CreateResource(name string, args ...interface{}) (*transit.DynamicMonitored
 			return nil, fmt.Errorf("unsupported value type: %T", reflect.TypeOf(arg))
 		}
 	}
-	// TODO: trickle up calculation from services?
+
+	if resource.Services != nil && len(resource.Services) != 0 {
+		resource.Status = CalculateResourceStatus(resource.Services)
```

Comment:
Why is the CalculateResourceStatus method empty? What is the difference between CalculateResourceStatus and CalculateStatus?

Reply:
Because you provided logic only for SNMP Connector, that's why I just added a skeleton of using status calculation function for another connectors

Reply:
OK, seems like other connectors have their own logic